### PR TITLE
editoast: move `BoundingBox` and `Zone` to `editoast_schemas`

### PR DESCRIPTION
**Pull request by hamz2a**
Part of #7048

### CODE REVIEW

File: editoast/editoast_schemas/src/primitives.rs [hunk 1]
```diff
@@ -1,13 +1,21 @@
 use derivative::Derivative;
 use enum_map::Enum;
+use geos::geojson::Geometry;
+use geos::geojson::Value::LineString;
+use geos::geojson::{self};
```
Comment (woshilapin) [inline editoast/editoast_schemas/src/primitives.rs]:
```suggestion
use geos::geojson;
```

**Review by woshilapin — APPROVED** [review]
Looks good overall, but let's wait on @leovalais's review because I'm not sure to understand the `inventory::submit` part.

File: editoast/Cargo.toml [hunk 2]
```diff
@@ -69,7 +70,7 @@ opentelemetry-datadog = { version = "0.10.0", features = ["reqwest-client"] }
 opentelemetry-otlp = "0.15.0"
 opentelemetry-semantic-conventions = "0.14.0"
 opentelemetry_sdk = { version = "0.22.1", features = ["rt-tokio", "trace"] }
-osm4routing = "0.6.1"
+osm4routing.workspace = true
```
Comment (leovalais) [inline editoast/Cargo.toml]: This crate doesn't belong in schema as it's OpenStreetMap related. We're only using it for a simple calculation in BoundingBox so I'd suggest to just inline the crate's function in editoast_schemas. Add a few lines of doc explaining what it does and why we're not using osm4routing (and credits Tristram for the implem 😁).

It's a shame that geos doesn't have this built in though.

File: editoast/editoast_schemas/Cargo.toml [hunk 3]
```diff
@@ -8,8 +8,11 @@ license = "LGPL-3.0"
 derivative.workspace = true
 editoast_common.workspace = true
 enum-map.workspace = true
+geos.workspace = true
```
Comment (leovalais) [inline editoast/editoast_schemas/Cargo.toml]: We don't use `geos` features in schemas, only its re-export of the crate `geojson`. I'd suggest you use `geojson` directly in schema to keep the dependency tree as small as possible.

**Review by leovalais — CHANGES_REQUESTED** [review]
Thanks! A few dependency issues.

@woshilapin We use `inventory` to register each error type that the frontend might receive and export them in our OpenApi. This way the frontend can have all their keys and translate them. Until now, all those `inventory::submit!` were generated by the `EditoastError` derive macro. So this is fine for now, but we'll need to find a solution later because we can't manually add that for each error we have.

File: editoast/editoast_schemas/Cargo.toml (hunk 3 above)
Comment (hamz2a) [reply editoast/editoast_schemas/Cargo.toml]: Done.

File: editoast/Cargo.toml (hunk 2 above)
Comment (hamz2a) [reply editoast/Cargo.toml]: Done.

File: editoast/editoast_schemas/Cargo.toml [hunk 4]
```diff
@@ -8,8 +8,10 @@ license = "LGPL-3.0"
 derivative.workspace = true
 editoast_common.workspace = true
 enum-map.workspace = true
+geojson = "0.24.1"
```
Comment (leovalais) [inline editoast/editoast_schemas/Cargo.toml]: Just define it in the workspace Cargo.toml with a relaxed version in order to let geos use the one it likes.

File: editoast/editoast_schemas/src/primitives.rs [hunk 5]
```diff
@@ -81,3 +89,143 @@ impl ObjectRef {
...
+        // Earth's mean radius in meters
+        let r: f64 = 6_378_100.0;
+
+        // Calculate differences in longitude and latitude in radians
+        let d_lon: f64 = (self.1 .0 - self.0 .0).to_radians();
+        let d_lat: f64 = (self.1 .1 - self.0 .1).to_radians();
+
+        // Convert latitude to radians
+        let lat1: f64 = (self.0 .1).to_radians();
+        let lat2: f64 = (self.1 .1).to_radians();
```
Comment (leovalais) [inline editoast/editoast_schemas/src/primitives.rs]: Better explicitly name the variables instead of `self.0.1`

File: editoast/src/error.rs [hunk 6]
```diff
@@ -220,6 +220,30 @@ impl EditoastError for json_patch::PatchError {
     }
 }
 
+inventory::submit! {
+    crate::error::ErrorDefinition::new("editoast:geometry:UnexpectedGeometry", "UnexpectedGeometry", "GeometryError", 404u16, "{\"expected\":\"String\",\"actual\":\"String\"}")
```
Comment (leovalais) [inline editoast/src/error.rs]: You can use an `r#""#` literal to avoid escaping

**Review by leovalais — CHANGES_REQUESTED** [review]
A few more minor changes

**Review by leovalais — APPROVED** [review]
LGTM thanks!

**Pull request merged**